devon4ng-architecture/index.asciidoc

```diff
@@ -0,0 +1,60 @@
+= Understanding the architecture principles followed by devon4ng
+====
+## Prerequisites
+
+Basic Angular knowledge
+
+## Learning goals
+
+This tutorial aims to explain concepts and principles of devon4ng architecture by refering an Angular application template. Below are the steps we will go through to understand it clearly.
+
+* Understand Angular architectural terminology (components and modules)
+* Different layers of an Angular application
+* Clone and understand a devon4ng application structure 
+
+By end of this tutorial you will have detailed understanding of principles followed by devon4ng application as well as its structure and architecture. 
+
+====
+
+[step]
+--
+restoreDevonfwIde(["npm", "ng", "vscode"])
+--
+
+[step]
+--
+nextKatacodaStep("Overview", [{ "file": "files/overview.md" }])
+--
+
+[step]
+--
+nextKatacodaStep("Architectural Layers", [{ "image": "images/architecture-layers.png" }, { "file": "files/layers.md" }])
+--
+
+[step]
+--
+nextKatacodaStep("Modules", [{ "image": "images/architecture-modules.png" }, { "file": "files/modules.md" }])
+--
+
+devon4ng-application-template is a simple application containing a home page (with some dummy text) and a data page (with CRUD operations). After cloning you can checkout the file structure and understand the concepts explained earlier.
+[step]
+== Cloning the reference application
+--
+cloneRepository("", "https://github.com/devonfw/devon4ng-application-template.git")
```

Comment:
Wouldn't it be better to clone the repo first and then explain modules, layers,.. using this as an example? Otherwise, the user only reads and does not see how these concepts are looking in real life.

Reply:
@SchettlerKoehler I did try that approach earlier, but did not like the idea to clone the repo without any explanation on the concepts.  Now I have retained the initial "Overview" step and moved the git cloning to second step from the last step.
Let me know how it looks now.

Reply:
I would make the cloning a setup script so the user does not have to do this.

Reply:
@SchettlerKoehler Done. Though I added the link to the git repo of the app we are cloning in the tutorial, incase someone wants to check it out. Hope it looks good now.